plugins/utils/navbuddy.nix

```
@@ -0,0 +1,273 @@
+          '';
+          branch = helpers.defaultNullOpts.mkStr "  " ''
+            The icon to use for branch nodes.
+          '';
+        };
+      };
+
+      icons =
+        mapAttrs
+        (
+          name: default:
+            helpers.defaultNullOpts.mkStr default "icon for ${name}."
+        )
+        {
+          File = "󰈙 ";
+          Module = " ";
+          Namespace = "󰌗 ";
+          Package = " ";
+          Class = "󰌗 ";
+          Method = "󰆧 ";
+          Property = " ";
+          Field = " ";
+          Constructor = " ";
+          Enum = "󰕘";
+          Interface = "󰕘";
+          Function = "󰊕 ";
+          Variable = "󰆧 ";
+          Constant = "󰏿 ";
+          String = "󰀬 ";
+          Number = "󰎠 ";
+          Boolean = "◩ ";
+          Array = "󰅪 ";
+          Object = "󰅩 ";
+          Key = "󰌋 ";
+          Null = "󰟢 ";
+          EnumMember = " ";
+          Struct = "󰌗 ";
+          Event = " ";
+          Operator = "󰆕 ";
+          TypeParameter = "󰊄 ";
+        };
+
+      useDefaultMapping = helpers.defaultNullOpts.mkBool true ''
+        If set to false, only mappings set by user are set. Else default mappings are used for keys that are not set by user
+      '';
+
+      keymapsSilent = mkOption {
+        type = types.bool;
+        description = "Whether navbuddy keymaps should be silent";
+        default = false;
+      };
+
+      mappings =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            attrsOf
+            (
+              either
+              str
+              helpers.rawTypes
+            )
+        )
+        ''
+          {
+            "<esc>" = "close";
+            "q" = "close";
+            "j" = "next_sibling";
+            "k" = "previous_sibling";
+
+            "h" = "parent";
+            "l" = "children";
+            "0" = "root";
+
+            "v" = "visual_name";
+            "V" = "visual_scope";
+
+            "y" = "yank_name";
+            "Y" = "yank_scope";
+
+            "i" = "insert_name";
+            "I" = "insert_scope";
+
+            "a" = "append_name";
+            "A" = "append_scope";
+
+            "r" = "rename";
+
+            "d" = "delete";
+
+            "f" = "fold_create";
+            "F" = "fold_delete";
+
+            "c" = "comment";
+
+            "<enter>" = "select";
+            "o" = "select";
+            "J" = "move_down";
+            "K" = "move_up";
+
+            "s" = "toggle_preview";
+
+            "<C-v>" = "vsplit";
+            "<C-s>" = "hsplit";
+          }
+        ''
+        ''
+           Actions to be triggered for specified keybindings. It can take either action name i.e `toggle_preview`
+          Or it can a `rawType`.
+        '';
+
+      lsp = {
+        autoAttach = helpers.defaultNullOpts.mkBool false ''
+          If set to true, you don't need to manually use attach function
+        '';
+
+        preference = helpers.defaultNullOpts.mkListStr null ''
```

```suggestion
        preference = helpers.mkNullOrOption (with types; listOf str) ''
```
When options have `null` as default, use `mkNullOrOption` instead of `defaultNullOpts.mkFoooo`.
Furthermore, `mkListStr` does not exist.
It's always good to run `nix flake check` before pushing so that you make sure that most of those errors are fixed :)

---

I did see the error message but didnt understand it, so I just pushed the changes so it was on github at least :sweat:.

Thanks for all the help!!!!

---

No worries ! Getting it our convoluted workflow is not easy, it is normal to make mistakes :)